### PR TITLE
Implemented copy feature

### LINKED ISSUE #1299: Copy Response Feature
## :warning: Is your feature request related to a problem? Please describe

There is no way to copy the response 

## :bulb: Describe the solution you'd like

Added a copy button in UI to copy the response

## :page_facing_up: Additional context

Screenshot_1728988025
Screenshot_1728988040

## :raising_hand: Do you want to develop this feature yourself?

- [x] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by A7ak**
## :camera: Screenshots

Screenshot_1728988025
Screenshot_1728988040

## :page_facing_up: Context

Fixes #1299

## :pencil: Changes

Added a button to copy the response

### CODE REVIEW

**Review by shivanandyadav11 — APPROVED** [review]

**Review by shivanandyadav11 — APPROVED** [review]

File: library/src/main/kotlin/com/chuckerteam/chucker/internal/ui/transaction/TransactionPayloadFragment.kt [hunk 1]
```diff
@@ -151,6 +153,22 @@ internal class TransactionPayloadFragment :
         }
     }
 
+    private fun copyResponse() {
```
Comment (cortinico) [inline library/src/main/kotlin/com/chuckerteam/chucker/internal/ui/transaction/TransactionPayloadFragment.kt:156]: Can we make sure the copy works for both request and response?

File: library/src/main/res/layout/chucker_transaction_item_copy.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<LinearLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:orientation="horizontal"
+    android:gravity="end">
+
+    <ImageButton
```
Comment (cortinico) [inline library/src/main/res/layout/chucker_transaction_item_copy.xml:8]: Can we tint this as the rest of the text?
As it looks weird that this icon is dark black while the rest of the text is not

File: library/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -70,4 +70,6 @@
     <string name="chucker_graphql_operation_is_empty"> &lt;Unable to discover GraphQL operation name&gt;</string>
     <string name="chucker_scroll_buttons_for_search">Buttons to scroll the search items</string>
     <string name="chucker_search_results_title">Search Results:</string>
+    <string name="chucker_copy_response">Copy Response</string>
+    <string name="chucker_response_copied">Response Copied</string>
```
Comment (cortinico) [inline library/src/main/res/values/strings.xml:74]: Those should be adapted for request/response

**Review by cortinico — COMMENTED** [review]
Sorry for the late review, but I was quite busy.
Code looks good but there are a minor changes to include

File: library/src/main/res/values/strings.xml (hunk 3 above)
Comment (A7ak) [reply library/src/main/res/values/strings.xml:74]: changes done

File: library/src/main/res/layout/chucker_transaction_item_copy.xml (hunk 2 above)
Comment (A7ak) [reply library/src/main/res/layout/chucker_transaction_item_copy.xml:8]: changes done 
Screenshot_1732956576

**Comment by A7ak** [comment]
@cortinico Changes are done please review it. Thanks

File: library/src/main/kotlin/com/chuckerteam/chucker/internal/ui/transaction/TransactionPayloadFragment.kt [hunk 4]
```diff
@@ -151,6 +153,45 @@ internal class TransactionPayloadFragment :
...
+                    )
+                }
+            }
+        }
+    }
+
+    private fun copyToClipboard(
+        payload: String,
+        payloadType: String,
+        payloadTypeCopied: String,
```
Comment (cortinico) [inline library/src/main/kotlin/com/chuckerteam/chucker/internal/ui/transaction/TransactionPayloadFragment.kt:185]:
```suggestion
        toastSuccessMessage: String,
```

File: library/src/main/kotlin/com/chuckerteam/chucker/internal/ui/transaction/TransactionPayloadAdapter.kt [hunk 5]
```diff
@@ -24,7 +25,8 @@ import com.chuckerteam.chucker.internal.support.indicesOf
  * We're using a [RecyclerView] to show the content of the body line by line to do not affect
  * performances when loading big payloads.
  */
-internal class TransactionBodyAdapter : RecyclerView.Adapter<TransactionPayloadViewHolder>() {
+internal class TransactionBodyAdapter(private val onCopyResp: () -> Unit) :
```
Comment (cortinico) [inline library/src/main/kotlin/com/chuckerteam/chucker/internal/ui/transaction/TransactionPayloadAdapter.kt:28]:
```suggestion
internal class TransactionBodyAdapter(private val onCopyBodyListener: () -> Unit) :
```

File: library/src/main/kotlin/com/chuckerteam/chucker/internal/ui/transaction/TransactionPayloadAdapter.kt [hunk 6]
```diff
@@ -188,6 +198,20 @@ internal sealed class TransactionPayloadViewHolder(view: View) : RecyclerView.Vi
         }
     }
 
+    internal class CopyViewHolder(
+        private val copyBinding: ChuckerTransactionItemCopyBinding,
+        private val onCopyResp: () -> Unit,
```
Comment (cortinico) [inline library/src/main/kotlin/com/chuckerteam/chucker/internal/ui/transaction/TransactionPayloadAdapter.kt:203]: Same let's rename it to `onCopyBodyListener`

**Review by cortinico — COMMENTED** [review]
Code looks good @A7ak only a couple nits.
There is a problem with the icon in dark mode though (see screenshot below). The icon should be transparent while it's not. Plus I feel we should make it `#ffffff` as the icon on the top bar.

Screenshot_20241207-222417

Screenshot 2024-12-07 at 22 32 59

Could you look into it? After this we can merge it 👍

**Comment by A7ak** [comment]
@cortinico  requested changes done please review it.

**Review by cortinico — APPROVED** [review]

**Comment by A7ak** [comment]
Thanks, @cortinico learned a lot while working on the copy feature.☺️

**Pull request merged**